### ISSUE TITLE
[21] Identifying High Consideration E-Commerce Search Queries

### ISSUE THREAD

**Issue by noparkee** [noparkee/paper#17]
<img width="646" alt="스크린샷 2024-12-27 오전 1 43 38" src="https://github.com/user-attachments/assets/c238ef34-a034-4385-9f09-017e10407f74" />

[paper](https://arxiv.org/abs/2410.13951)

---

- High Consideration queries를 식별하는 task
- product search 시 query-related 쇼핑 지식 컨텐츠에 대한 잠재적 참여 수준을 나타내는 query ranking에 초점을 맞춘 Engagement-based Query Ranking (EQR) 제시
- main cotribution
- 참여도 지표를 기반으로 HC 쿼리의 순위를 매기는 Engagement-based Query Ranking 제시
- 향후 잠재적인 참여로 이어질 수 있는 새로운 쿼리를 효과적으로 식별할 수 있는 간단하고 효과적인 EQR 방법 제안 
- 오프라인 실험 결과, 제안한 방법이 모든 지표에서 EQR의 모든 기준선을 뛰어넘으며 human evaluation 결과 모델의 HC 쿼리 선택 precision이 96%.
- 상업적으로 배포한 결과 선택된 HC 쿼리의 downstream customer 영향이 human annotator가 선택한 쿼리보다 큰 것으로 나타남.

>high consideration query란?
 a search term that indicates a customer is making a careful and involved decision, and is likely to invest a significant amount of time and effort into research

> 본 논문에서는 유저가 QA 구성 요소의 관련 콘텐츠로부터 혜택을 받을 수 있는 새로운 쿼리를 식별하는 데 중점을 두고 있음

---
### Classification and Ranking Approaches
 결국 어떤 기준에 따라 쿼리가 HCQ인지 아닌지를 판별하는 것이기 때문에 분류 문제로 볼 수 있지만 분류 문제는 너무 단순해서 랭킹 문제로 접근하겠다.

#### Shortcomings of Binary Classification
- 단순한 분류 방식으로는 가장 promising한 쿼리를 먼저 처리해서 얻을 수 있는 이익을 극대화 할 수 없음.
- 또한, threshold 선택도 해야하고, 모델의 probability output이 잘 calibrated 됐는지도 확인해야함
- 위 두 가지 문제를 랭킹 방식으로는 피할 수 있음

#### Ranking Approach
- 목표는 상대적인 순위가 아니라 쿼리를 독립적으로 점수화하는 것이기 때문에 query performance prediction 작업과 유사한 pointwise ranking task 모델
- pairwise comparison 데이터가 아니라 절대적인 engagement 점수로 학습할 수 있음 => regression 접근법 가능
- pointwise 접근 방식이 해석이 더 쉽고, 새로운 쿼리에 대해서도 더 쉽게 추론 할 수 있음
- engagement 데이터를 기반으로 학습하기 때문에 incremental learning에도 적합.

### Pointwise Ranking Model
- proxy target, 쿼리 q의 engagement score $e$
  <img width="165" alt="스크린샷 2024-12-28 오후 6 02 23" src="https://github.com/user-attachments/assets/8ff092c7-79a1-44cc-96f2-4495c5c7fbb1" />

- $freq_c(q)$는 informational component를 클릭한 횟수이고 $freq(q)$는 쿼리의 전체 frequency
- 쿼리 $q \in Q$ 에 대해 사용자 interaction feature $x \in \mathcal{R}^d$ (d는 feature의 총 개수)
- query 의 feature가 주어졌을 때 query의 engagement score를 예측할 수 있는 함수 $f(-)$를 학습하는 것이 목표
- n개의 쿼리에 대해 predicted engagement score를 얻은 다음 내림차순으로 re-rank
- embedding을 사용하지 않음
  - query space가 크고, 쿼리 간에 미묘한 차이 (airpods vs. airpods case) 가 있을 수 있어서 일반화가 잘 될 수도 있음  - 임베딩 기반은 쿼리의 텍스트에 의존하지 않는 방법에 비해 성능이 떨어짐 (Section 5)
  - 텍스트 인코더는 학습 및 inference 모두 느림
  - behavioral features를 이용해 다국어 인코더를 사용하지 않아도 언어에 구애받지 않음

#### Training Data Acquisition
- 이전에 target되지 않은 쿼리에 대해 일반화할 수 있는 모델을 학습하기 위해 engagement score를 가진 작지만 대표적인 시드 쿼리 셋이 필요
  - 일반적으로 이 시드 쿼리 셋은 전문가가 manually 선정하고 커버리지를 위해 제품 카테고리에 따라 계층화

### Query Features
<img width="419" alt="스크린샷 2024-12-28 오후 7 54 57" src="https://github.com/user-attachments/assets/d25924a6-f135-4a59-ad17-4ccd672e06bc" />

#### Behavioral Features
- 쿼리에 대한 검색 시스템과 유저의 상호작용을 특징짓는 요소
- 쿼리가 제출된 뒤, 검색 결과를 클릭하거나 상품을 장바구니에 추가하는 등 후속 상호 작용 관찰

#### Financial Features
- financial signals
  - e.g. 주문량, 가격, temporal patterns

#### Catalog Features
- 검색 결과에 제공되는 상품의 카탈로그 기반 feature들

### Training
- EQR task를 pointwise regression task로 접근하며 query engagement score를 예측하기 위해 GBDT를 학습 시킴
- MSE 최소화
- $x_i$는 쿼리 $q_i$의 feature vector
  <img width="241" alt="스크린샷 2024-12-28 오후 11 01 48" src="https://github.com/user-attachments/assets/a02f7ac9-7e29-4c87-ba47-0dea8ac1394e" />

---

## Experimental Setup

- 1년 동안 아마존에서 QA 구성 요소와 관련된 데이터 수집
- 11273개의 쿼리가 manually 선택되어 QA 구성 요소를 트리거하고 curatetd contents를 표시
- proxy target으로 해당 query-level user engagement data 수집
- train:valid:test = 70:15:15

### Evaluation Metrics
- HIT@k: ground-truth engagement score가 가장 높은 k개의 쿼리를 postiive로 간주하여 모델이 예측한 상위 k개의 결과 중 positive query의 비율
- Kendall's Tau: prediction 리스트와 ground-truth 리스트에 대한 ordinal rank 상관계수. [-1, 1] 범위로 값이 클수록 유사성 높음
- MSE: ground-truth engagement와 prediction 사이의 mse (계절적 변화로 인해 변동될 수 있는 정확한 engagement 수준을 파악해야하기 때문에 더 어려운 지표(?))

### Baselines
- Frequency: 쿼리 빈도에 따라 순위가 매겨짐
- Regression methods: 위의 feature를 이용해 random forest, linear model (Lasso, Ridge, Elastic Net and linear regression)
- RoBERTa
- GPT-3.5 and GPT-4o (zero-shot)
- GPT-3.5 and GPT-4o (few-shot)
- RoBERTa와 GPT는 텍스트 임베딩만 사용했고 나머지는 non-text feature만 사용

----

## Results
<img width="672" alt="스크린샷 2024-12-28 오후 11 17 44" src="https://github.com/user-attachments/assets/70caf423-11ea-4185-9af1-219754f31336" />

### High Consideration Query Prediction
- Decision Tree Ensembles yield best overall  performance.
- We achieve high recall for the top queries.
- Ranking over the entire test set is accurate.
- Text-based methods underperform feature-based models.
  - query semantic 만으로는 이 task를 해결하기엔 부족. behavior features가 더 강한 cue를 제공

### Ablation Study and Feature Importance
<img width="858" alt="스크린샷 2024-12-28 오후 11 26 37" src="https://github.com/user-attachments/assets/b53535f6-fedd-47a3-a485-cc2de06da2cf" />

### Human Evaluation
- 모델을 사용하여 large unseen traffic을 예측 -> 예측된 score에서 가장 높은 순위와 가장 낮은 순위의 쿼리를 골고루 선택 => 1500개의 쿼리 셋
- 전문가가 셔플된 셋에 대해 HC인지 아닌지 분류 => 모델의 정확도는 96%
- 잠재적으로 annotator를 대체할 수 있는 능력!

---

### Commercial Deployment
- 모델이 선택한 쿼리가 사람이 선택한 쿼리보다 더 좋은 결과 (relative increase of 6%)
- 모델을 완전히 상용화 함으로써 human annotator의 필요성 제거 + HC selection application을 쉽게 확장할 수 있음

**Comment by noparkee** [comment]
- 이 논문은 딱히 llm과 관련 있지는 않음.
- 다만 쿼리의 semantic similarity 보다도 다른 다양한 feature가 중요하다! 는 정도의 의미를 가질 듯 (물론 태스크 마다 다르겠지만)
